src/ingestion/models/ingestionManager.ts

```diff
@@ -173,6 +176,133 @@ export class IngestionManager {
+      this.logger.error({ msg: message, logContext: logCtx, jobId, currentStatus: status, validStatuses });
+      const error = new BadRequestError(message);
+      trace.getActiveSpan()?.setAttribute('exception.type', error.status);
+      throw error;
+    }
+  }
+
+  @withSpanAsyncV4
+  private async getValidationTask(jobId: string, logCtx: LogContext): Promise<ITaskResponse<ValidationTaskParameters>> {
+    const tasks = await this.jobManagerWrapper.getTasksForJob<ValidationTaskParameters>(jobId);
+
+    const validationTask = tasks.find((task) => task.type === this.validationTaskType);
+
+    if (!validationTask) {
+      const message = `Cannot retry job with id: ${jobId} because no validation task was found`;
+      this.logger.error({ msg: message, logContext: logCtx, jobId, taskTypes: tasks.map((t) => t.type) });
+      const error = new NotFoundError(message);
+      trace.getActiveSpan()?.setAttribute('exception.type', error.status);
+      throw error;
+    }
+
+    return validationTask;
+  }
+
+  @withSpanAsyncV4
+  private async handleRetryWithoutErrors(jobId: string, taskId: string, logCtx: LogContext): Promise<ResponseId> {
+    this.logger.info({ msg: 'validation completed without errors, resetting job', logContext: logCtx, jobId, taskId });
+    await this.jobManagerWrapper.resetJob(jobId);
+    trace.getActiveSpan()?.setStatus({ code: SpanStatusCode.OK }).addEvent('ingestionManager.retryLayer.success', { retryType: 'reset', jobId });
+    this.logger.info({ msg: 'job reset successfully', logContext: logCtx, jobId, taskId });
+    return { jobId, taskId };
+  }
+
+  @withSpanAsyncV4
+  private async handleRetryWithErrors(
+    jobId: string,
+    retryJob: { parameters: z.infer<typeof ingestionBaseJobParamsSchema> },
+    validationTask: ITaskResponse<ValidationTaskParameters>,
+    logCtx: LogContext
+  ): Promise<ResponseId> {
+    this.logger.info({ msg: 'validation has errors, checking for shapefile changes', logContext: logCtx, jobId, taskId: validationTask.id });
+
+    const metadataShapefilePath = retryJob.parameters.inputFiles.metadataShapefilePath;
+    if (!metadataShapefilePath) {
+      const message = `Cannot retry job with id: ${jobId} because metadata shapefile path is missing from job parameters`;
+      this.logger.error({ msg: message, logContext: logCtx, jobId });
+      const error = new BadRequestError(message);
+      trace.getActiveSpan()?.setAttribute('exception.type', error.status);
+      throw error;
+    }
+
+    const absoluteMetadataPath = join(this.sourceMount, metadataShapefilePath);
+    const newChecksums = await this.getFilesChecksum(absoluteMetadataPath);
+
+    this.validateShapefileChanges(jobId, validationTask, newChecksums, logCtx);
+
+    const newTaskParameters: IUpdateTaskBody<ValidationTaskParameters> = {
+      parameters: { ...validationTask.parameters, checksums: [...validationTask.parameters.checksums, ...newChecksums] },
```

Comment:
you should only modify the changed checksum detected else its will duplicate the other same checksum for unchanged files